pkg/postgresql/postgresql.go

```diff
@@ -713,21 +713,10 @@ func (p *Manager) writePgHba() error {
 
 	// Minimal entries for local normal and replication connections needed by the stolon keeper
 	// Matched local connections are for postgres database and suUsername user with md5 auth
-	// Matched local replicaton connections are for replUsername user with md5 auth
+	// Matched local replication connections are for replUsername user with md5 auth
 	f.WriteString(fmt.Sprintf("local postgres %s %s\n", p.suUsername, p.suAuthMethod))
 	f.WriteString(fmt.Sprintf("local replication %s %s\n", p.replUsername, p.replAuthMethod))
 
-	// By default accept all connections for the superuser suUsername with md5 auth
-	// Used for pg_rewind resyncronization
-	// TODO(sgotti) Configure this dynamically based on our followers provided by the clusterview
-	f.WriteString(fmt.Sprintf("host all %s %s %s\n", p.suUsername, "0.0.0.0/0", p.suAuthMethod))
-	f.WriteString(fmt.Sprintf("host all %s %s %s\n", p.suUsername, "::0/0", p.suAuthMethod))
-
-	// By default accept all replication connections for the replication user with md5 auth
-	// TODO(sgotti) Configure this dynamically based on our followers provided by the clusterview
-	f.WriteString(fmt.Sprintf("host replication %s %s %s\n", p.replUsername, "0.0.0.0/0", p.replAuthMethod))
-	f.WriteString(fmt.Sprintf("host replication %s %s %s\n", p.replUsername, "::0/0", p.replAuthMethod))
-
 	if p.hba != nil {
 		for _, e := range p.hba {
 			f.WriteString(e + "\n")
```

Comment:
You have forgot to handle the "else" case below inside generateHBA. Here it'll always be skipped since p.hba is never empty.

Reply:
Ok, this means that previously, when p.hba was empty, the generated hba was:

```
host replication repluser 0.0.0.0/0 md5
host replication repluser ::0/0 md5
host all superuser 0.0.0.0/0 md5
host all superuser ::0/0 md5
host all all 0.0.0.0/0 md5
host all all ::0/0 md5
```
which is same as this:
```
host all all 0.0.0.0/0 md5
host all all ::0/0 md5
```
but simpler so I think this should be the result in "all" access mode.

In "strict" access mode we should not use this rule.

Reply:
Not always, the first lines can have a different auth method than md5 (currently only trust but in future other methods could be added), while the last lines are added only if there's no user provided pgHBA and always have md5. I'd prefer to not change the current logic.

Reply:
Yes you're right ! 

For the strict mode, I still would prefer not to add this rule.

Reply:
It may be confusing but these are two separate things. The first lines (the one affected by this patch to set the DefaultSuReplAccessMode) are for instance replication, the latter is a default when the user doesn't provide a PGHBA. So I'll leave them distinct and not make DefaultSuReplAccessMode change this.

Reply:
Ok I understand !
Can I plan a DefaultUserAccessMode feature ?

Reply:
@t0k4rt you can just do the same proving an empty (not nil) pgHBA in the clusterspec